wagtail/contrib/modeladmin/options.py

```diff
@@ -243,13 +243,16 @@ def get_ordering(self, request):
 
     def get_queryset(self, request):
         """
-        Returns a QuerySet of all model instances that can be edited by the
-        admin site.
+        Returns a QuerySet of all model instances that can be explored by the
+        current user.
         """
         qs = self.model._default_manager.get_queryset()
+        qs = filter_explorable_pages(qs, request.user, include_ancestors=False)
```

Comment:
Given that the primary motivation for ModelAdmin's page views is to present the page listing in a way that doesn't require navigating the tree, do we really want to filter to the explorable pages? It seems to me it would be more appropriate to narrow it down to the pages that the user has edit permission for.

(If so - and modeladmin doesn't currently filter the listing by user permission - then that would probably be a job for a separate PR, as this change would then be independent of the task of implementing 'explore' permission restrictions.)